### PR TITLE
Update _index.md

### PR DESCRIPTION
**Pull request by justinegeffen**
Fixed typos and grammar.

### CODE REVIEW

File: site/content/integrate/apps/_index.md [hunk 1]
```diff
@@ -33,20 +33,20 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
 
 ### When would you build an app vs a custom slash command vs a webhook vs a plugin?
 
-That depends on your use case, as they each have benefits. Apps can be advertised, shared, and installed by others via the Marketplace. Comparatively, a custom slash command requires some configuration and is less interactive.
+That depends on your use case, as they each have benefits.
 
-The built-in incoming webhook is rigid and requires specific fields to be posted to it's URL, but some external systems don't allow the customization of the fields included in the webhook payload. Compared to built-in webhooks, An App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
+The built-in incoming webhook is rigid and requires specific fields to be posted to its URL - but some external systems don't allow the customization of the fields included in the webhook payload. Compared to built-in webhooks, an App can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:38]: As pointed out in https://github.com/mattermost/mattermost-developer-documentation/pull/804#discussion_r622019565 I think need to emphasize more the one rice UX on webapp and _mobile_. That is something that plugins can't do.

Also, it's easier to develop an app then a plugin.

**Review by hanzei — CHANGES_REQUESTED** [review]
Should we combine the two questions into one and lay down when you want to use what?

File: site/content/integrate/apps/_index.md [hunk 2]
```diff
@@ -33,20 +33,20 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
 
-A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
+A plugin should be used when you need to alter the UX in Mattermost or have an application that requires low latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
```
Comment (mickmister) [inline site/content/integrate/apps/_index.md:40]:
```suggestion
A plugin should be used when you need to [directly alter the UI](https://developers.mattermost.com/extend/plugins/webapp/best-practices) in Mattermost or you have a feature that requires low latency with the server (such as replacing characters in any message [before it is saved](https://developers.mattermost.com/extend/plugins/server/reference/#Hooks.MessageWillBePosted)). Currently, plugins have several [UX hooks](https://developers.mattermost.com/extend/plugins/server/reference/#Hooks) that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time. Please see the [plugin documentation](https://developers.mattermost.com/extend/plugins) for more information.
```

File: site/content/integrate/apps/_index.md (hunk 1 above)
Comment (mickmister) [reply site/content/integrate/apps/_index.md:38]:
```suggestion
The built-in incoming webhook is great for simple use cases. It requires the incoming payload to contain a valid [Post](https://pkg.go.dev/github.com/nhannv/mattermost-server/model#Post) JSON object - but some external systems don't allow the customization of the fields included in the webhook payload. Compared to built-in webhooks, an App or plugin can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
```

**Review by mickmister — CHANGES_REQUESTED** [review]
Change some words and added some links. Let me know what you think 🙂

**Comment by mickmister** [comment]
Do we also want to mention [outgoing webhooks](https://docs.mattermost.com/developer/webhooks-outgoing.html)?

File: site/content/integrate/apps/_index.md [hunk 3]
```diff
@@ -41,20 +41,20 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
 
-A plugin should be used when you need to alter the React UX in Mattermost or have an application that requires low-latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
+A plugin should be used when you need to [directly alter the UI](https://developers.mattermost.com/extend/plugins/webapp/best-practices) in Mattermost or you have a feature that requires low latency with the server (such as replacing characters in any message [before it is saved](https://developers.mattermost.com/extend/plugins/server/reference/#Hooks.MessageWillBePosted)). Currently, plugins have several [UX hooks](https://developers.mattermost.com/extend/plugins/server/reference/#Hooks) that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time. Please see the [plugin documentation](https://developers.mattermost.com/extend/plugins) for more information.
```
Comment (hanzei) [inline site/content/integrate/apps/_index.md:48]: `ref` should be used as it finds invalid links:

```suggestion
A plugin should be used when you need to [directly alter the UI]({{< ref "extend/plugins/webapp/best-practices" >}}) in Mattermost or you have a feature that requires low latency with the server (such as replacing characters in any message [before it is saved]({{< ref "extend/plugins/server/reference#Hooks.MessageWillBePosted" >}})). Currently, plugins have several [UX hooks]({{< ref "extend/plugins/server/reference#Hooks" >}}) that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time. Please see the [plugin documentation]({{< ref "extend/plugins" >}}) for more information.
```

File: site/content/integrate/apps/_index.md [hunk 4]
```diff
@@ -33,20 +33,20 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
+A plugin should be used when you need to alter the UX in Mattermost or have an application that requires low latency with the server (such as replacing characters in any message). Currently, plugins have several UX hooks that Apps cannot access, however we plan to add/migrate more UX hooks into the Apps Framework over time.
 
 ### What's the difference between the apps framework and the plugin framework?
 
 The App framework provides a few differences from plugins.
 
 - We’ve made our interactive elements easier to use and develop.
 - Works on both desktop and mobile clients without any extra code.
-- You host the app code or build for Matterhost and distribute through the Marketplace.
+- You host the app code or build for Mattermost and distribute through the Marketplace.
```
[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/_index.md:56]: This is this not resolved.

**Review by hanzei — CHANGES_REQUESTED** [review]

File: site/content/integrate/apps/_index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (justinegeffen) [reply site/content/integrate/apps/_index.md:56]: It's resolved in a separate PR. :)

**Review by hanzei — APPROVED** [review]

File: site/content/integrate/apps/_index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/_index.md:56]: Which is it? I can't find it.

**Review by aaronrothschild — APPROVED** [review]

File: site/content/integrate/apps/_index.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (justinegeffen) [reply site/content/integrate/apps/_index.md:56]: Sorry - I was referring to Matterhost > Mattermost! 

Do you recommend just removing this line?

[the remark this answers is not in the document]
Comment (hanzei) [reply site/content/integrate/apps/_index.md:56]: I don't know what the original intent of the line was, but as it is, it's not correct and should be removed.

**Review by hanzei — APPROVED** [review]

**Comment by justinegeffen** [comment]
@hanzei, can you dismiss @mickmister's review? I have implemented the changes but I think he's offline and won't be able to approve in time. :)

**Comment by mickmister** [comment]
@justinegeffen I'm here :)

**Comment by mickmister** [comment]
Do we want to add an example for outgoing webhooks or can/should that be in a different PR?

**Comment by justinegeffen** [comment]
Oh awesome! I should learn to check stand-up. :)

**Comment by justinegeffen** [comment]
> Do we want to add an example for outgoing webhooks or can/should that be in a different PR?
Let's merge this and then add those in a new PR if that works for you? Do you have the examples available?

File: site/content/integrate/apps/_index.md [hunk 5]
```diff
@@ -41,20 +41,19 @@ Read the [JavaScript start guide]({{< ref "quick-start-js" >}}) or [Go quick sta
...
-- We’ve made our interactive elements easier to use and develop.
-- Works on both Mobile and Desktop clients without any extra code.
-- You host the app code, or build for Mattermost, and distribute through the Marketplace.
+- Interactive elements are easier to use and develop.
+- Compatible with both Mobile and Desktop clients without any extra code.
 
 ### What language should I use to write apps?
 
-Any language you want. However, the only language with an official driver right now is Go. We're planning to release other drivers in the future in conjunction with our community.
+Any language you want. We currently have an official driver for Go, and we are planning to release other drivers in the future in conjunction with our community.
```
Comment (mickmister) [inline site/content/integrate/apps/_index.md:59]: Maybe we want to link to the Go "driver"? Which would be the structs available at https://github.com/mattermost/mattermost-plugin-apps/tree/master/apps I think. Also, we are working on creating one for javascript cc @hanzei

Comment (mickmister) [reply site/content/integrate/apps/_index.md:59]: As well as any app-related utility functions used in our examples

**Review by mickmister — APPROVED** [review]

File: site/content/integrate/apps/_index.md (hunk 5 above)
Comment (mickmister) [reply site/content/integrate/apps/_index.md:59]: Maybe just link to the go quick start and say something like "See the driver's usage here"

**Pull request merged**